src/main/drivers/flash_w25q128fv.c

```diff
@@ -115,45 +119,71 @@ w25q128fvState_t w25q128fvState = { 0 };
 static bool w25q128fv_waitForReady(flashDevice_t *fdevice);
 static void w25q128fv_waitForTimeout(flashDevice_t *fdevice);
 
-static void w25q128fv_setTimeout(flashDevice_t *fdevice, uint32_t timeoutMillis)
+MMFLASH_CODE static void w25q128fv_setTimeout(flashDevice_t *fdevice, timeMs_t timeoutMillis)
 {
-    uint32_t now = HAL_GetTick();
-    fdevice->timeoutAt = now + timeoutMillis;
+    timeMs_t nowMs = microsISR() / 1000;
+    fdevice->timeoutAt = nowMs + timeoutMillis;
```

Comment:
`micros() / 1000` will behave terribly on overflow (you can set timeout to value outside timer/1000 range)
Better multiply by 1000 and use micros

And is this code really called only from ISR ?

Reply:
The wait/delay functions must work with interrupts disabled, so microsISR() is needed as millis() requires systick to be enabled.

See git blame on those lines which reveals: 2125ce14633b29cd4721735eea7e62bc2322f7d7